delocate/libsana.py

```diff
@@ -4,16 +4,194 @@
 """
 
 import os
-from os.path import basename, join as pjoin, realpath
-
+from os.path import basename, dirname, join as pjoin, realpath
+
+import logging
+from typing import (
+    Callable,
+    Dict,
+    Iterable,
+    Iterator,
+    List,
+    Optional,
+    Set,
+    Text,
+    Tuple,
+)
 import warnings
 
+import six
+
 from .tools import (get_install_names, zip2dir, get_rpaths,
                     get_environment_variable_paths)
 from .tmpdirs import TemporaryDirectory
 
 
-def tree_libs(start_path, filt_func=None):
+logger = logging.getLogger(__name__)
+
+
+class DependencyNotFound(Exception):
+    """Raised by tree_libs or resolve_rpath if an expected dependency is
+    missing.
+    """
```

Comment:
First line of docstring should be self-contained:

```suggestion
    """
    Raised by tree_libs or resolve_rpath if an expected dependency is missing.
    """
```